### PR TITLE
default single table rule routing and configuration support

### PR DESCRIPTION
**Pull request by lanchengx**
For https://github.com/apache/shardingsphere/issues/13582.

### CODE REVIEW

**Comment by RaigorJiang** [comment]
Hi @strongduanmu ,
please help to review the PR, thank you.

**Comment by strongduanmu** [comment]
> Hi @strongduanmu , please help to review the PR, thank you.

@RaigorJiang Thank you for your remind, I will review this pr.

File: shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/main/java/org/apache/shardingsphere/singletable/rule/SingleTableRule.java [hunk 1]
```diff
@@ -113,6 +117,14 @@ public boolean isAllTablesInSameDataSource(final RouteContext routeContext, fina
         return true;
     }
     
+    /**
+     * Get default data source.
```
Comment (strongduanmu) [inline shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/main/java/org/apache/shardingsphere/singletable/rule/SingleTableRule.java:121]: @lanchengx A blank line is required after javadoc.

File: shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/main/java/org/apache/shardingsphere/singletable/rule/builder/SingleTableRuleBuilder.java [hunk 2]
```diff
@@ -33,7 +33,9 @@
     
     @Override
     public SingleTableRule build(final SchemaRulesBuilderMaterials materials, final SingleTableRuleConfiguration config, final Collection<ShardingSphereRule> builtRules) {
-        return new SingleTableRule(materials.getDatabaseType(), materials.getDataSourceMap(), builtRules, materials.getProps());
+        SingleTableRule singleTableRule = new SingleTableRule(materials.getDatabaseType(), materials.getDataSourceMap(), builtRules, materials.getProps());
+        config.getDefaultDataSource().ifPresent(singleTableRule::setDefaultDataSource);
```
Comment (strongduanmu) [inline shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/main/java/org/apache/shardingsphere/singletable/rule/builder/SingleTableRuleBuilder.java:37]: @lanchengx Can we use the constructor to replace the setDefaultDataSource method?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Comment by lanchengx** [comment]
@strongduanmu It has been modified, please wait for a while CI.

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**